packages/x-charts/src/internals/plugins/corePlugins/useChartId/useChartId.ts

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { ChartPlugin } from '../../models';
+import { UseChartIdSignature } from './useChartId.types';
+import { createChartDefaultId } from './useChartId.utils';
+
+export const useChartId: ChartPlugin<UseChartIdSignature> = ({ params, store }) => {
+  React.useEffect(() => {
+    store.update((prevState) => {
+      if (params.id === prevState.id.providedChartId && prevState.id.chartId !== undefined) {
+        return prevState;
+      }
+
+      return {
+        ...prevState,
+        id: { ...prevState.id, chartId: params.id ?? createChartDefaultId() },
+      };
+    });
+  }, [store, params.id]);
+  return {};
+};
+
+useChartId.params = {
+  id: true,
+};
```

Comment:
where are the these params used? 🤔

Reply:
In the tree view, they are used to split the props that are forwarded to the DOM and the props that are owned by the plugins and should not be forwarded to the DOM